rosbag2_cpp/src/rosbag2_cpp/typesupport_helpers.cpp

```diff
@@ -105,33 +106,51 @@ get_typesupport(const std::string & type, const std::string & typesupport_identi
   std::string type_name;
   std::tie(package_name, middle_module, type_name) = extract_type_identifier(type);
 
-  std::string poco_dynamic_loading_error = "Something went wrong loading the typesupport library "
-    "for message type " + package_name + "/" + type_name + ".";
+  std::stringstream rcutils_dynamic_loading_error;
+  rcutils_dynamic_loading_error <<
+    "Something went wrong loading the typesupport library for message type " << package_name <<
+    "/" << type_name << ".";
 
   auto library_path = get_typesupport_library_path(package_name, typesupport_identifier);
 
-  try {
-    auto typesupport_library = std::make_shared<Poco::SharedLibrary>(library_path);
-
-    auto symbol_name = typesupport_identifier + "__get_message_type_support_handle__" +
-      package_name + "__" + (middle_module.empty() ? "msg" : middle_module) + "__" + type_name;
-
-    if (!typesupport_library->hasSymbol(symbol_name)) {
-      throw std::runtime_error(poco_dynamic_loading_error + " Symbol not found.");
+  std::unique_ptr<rcutils_shared_library_t> typesupport_library =
+    std::make_unique<rcutils_shared_library_t>();
+  *typesupport_library = rcutils_get_zero_initialized_shared_library();
+
+  rcutils_ret_t ret = rcutils_load_shared_library(typesupport_library.get(), library_path.c_str());
+  if (ret != RCUTILS_RET_OK) {
+    if (ret == RCUTILS_RET_BAD_ALLOC) {
+      throw std::runtime_error{rcutils_dynamic_loading_error.str() + std::string(
+                " failed to allocate memory")};
+    } else if (ret == RCUTILS_RET_INVALID_ARGUMENT) {
+      throw std::runtime_error{rcutils_dynamic_loading_error.str() + std::string(
+                " invalid arguments")};
+    } else {
+      throw std::runtime_error{
+              rcutils_dynamic_loading_error.str() + std::string(" Library could not be found ") +
+              library_path};
     }
+  }
```

Comment:
generally not a big fan of code duplication.

Can we factor this somewhat into a function within `typesupport_helpers.cpp` which then can also be used from the code snippet above?